cicecore/cicedyn/infrastructure/io/io_pio2/ice_history_write.F90

```diff
@@ -509,165 +478,96 @@ subroutine ice_write_hist (ns)
-      dimidz(4) = timid
-
-      do n = n2D + 1, n3Dccum
-         if (avail_hist_fields(n)%vhistfreq == histfreq(ns) .or. write_ic) then
-            call ice_pio_check(pio_def_var(File, trim(avail_hist_fields(n)%vname), lprecision, dimidz, varid), &
-                 subname//' ERROR: defining var'//trim(avail_hist_fields(n)%vname),file=__FILE__,line=__LINE__)
-            call ice_write_hist_attrs(File,varid,avail_hist_fields(n),ns)
-         endif
-      enddo  ! num_avail_hist_fields_3Dc
-
-      !-----------------------------------------------------------------
-      ! 3D (ice layers)
-      !-----------------------------------------------------------------
-
-      dimidz(1) = imtid
-      dimidz(2) = jmtid
-      dimidz(3) = kmtidi
-      dimidz(4) = timid
-
-      do n = n3Dccum + 1, n3Dzcum
-         if (avail_hist_fields(n)%vhistfreq == histfreq(ns) .or. write_ic) then
-            call ice_pio_check(pio_def_var(File, trim(avail_hist_fields(n)%vname), lprecision, dimidz, varid), &
-                 subname//' ERROR: defining var'//trim(avail_hist_fields(n)%vname),file=__FILE__,line=__LINE__)
-            call ice_write_hist_attrs(File,varid,avail_hist_fields(n),ns)
-         endif
-      enddo  ! num_avail_hist_fields_3Dz
-
-      !-----------------------------------------------------------------
-      ! 3D (biology ice layers)
-      !-----------------------------------------------------------------
-
-      dimidz(1) = imtid
-      dimidz(2) = jmtid
-      dimidz(3) = kmtidb
-      dimidz(4) = timid
-
-      do n = n3Dzcum + 1, n3Dbcum
-         if (avail_hist_fields(n)%vhistfreq == histfreq(ns) .or. write_ic) then
-            call ice_pio_check(pio_def_var(File, trim(avail_hist_fields(n)%vname), lprecision, dimidz, varid), &
-                 subname//' ERROR: defining var'//trim(avail_hist_fields(n)%vname),file=__FILE__,line=__LINE__)
-            call ice_write_hist_attrs(File,varid,avail_hist_fields(n),ns)
-         endif
-      enddo  ! num_avail_hist_fields_3Db
-
-      !-----------------------------------------------------------------
-      ! 3D (biology snow layers)
-      !-----------------------------------------------------------------
-
-      dimidz(1) = imtid
-      dimidz(2) = jmtid
-      dimidz(3) = kmtida
-      dimidz(4) = timid
-
-      do n = n3Dbcum + 1, n3Dacum
-         if (avail_hist_fields(n)%vhistfreq == histfreq(ns) .or. write_ic) then
-            call ice_pio_check(pio_def_var(File, trim(avail_hist_fields(n)%vname), lprecision, dimidz, varid), &
-                 subname//' ERROR: defining var'//trim(avail_hist_fields(n)%vname),file=__FILE__,line=__LINE__)
-            call ice_write_hist_attrs(File,varid,avail_hist_fields(n),ns)
-         endif
-      enddo  ! num_avail_hist_fields_3Da
-
-      !-----------------------------------------------------------------
-      ! 3D (fsd)
-      !-----------------------------------------------------------------
+        dimid3(1) = imtid
```

Comment:
Oh I thought they were kind of confusing - because they were all under the comment "! define attributes for time-variant variables" but then used the horizontal lines in the dame way that comment does.